Project 2/k-NNAlgorithms/src/MathFunction.java

```diff
@@ -27,4 +27,176 @@ public static double hammingDistance(ArrayList<String> in1, ArrayList<String> in
         }
         return result;
     }
+
+    public static String average (ArrayList<String> input){
+        double sum=0;
+        for (int i = 0; i <input.size() ; i++) {
+            sum+= Double.parseDouble(input.get(i));
+        }
+        sum/=input.size();
+        return sum+"";
+    }
+
+    public static String mode(ArrayList<String> input){
+        String mode="";
+        int maxIndex=0;
+        int currMax=0;
+        for (int i = 0; i <input.size() ; i++) {
+            int countOfClass = Collections.frequency(input,input.get(i));
+            if(countOfClass>=currMax){
+                maxIndex=i;
+            }
+        }
+        mode=input.get(maxIndex);
+        return mode;
+    }
+
+
+    public static ArrayList<String> processConfusionMatrix(ArrayList<String> results, ArrayList<ArrayList<String>> testData){
+
+        int lengthOfData = results.size();
+        int classIndex = testData.get(0).size()-1;
+
+        //initialize classifications in the confusion matrix as the first row
+        ArrayList<ArrayList<String>> confusionMatrix= new ArrayList<>();
+        confusionMatrix.add(new ArrayList<>());
+        for (int i = 0; i <lengthOfData ; i++) {
+            if(!confusionMatrix.get(0).contains(testData.get(i).get(classIndex))){
```

Comment:
Constant to clarify that 0 is used for classification column

Reply:
It's a row, but yes that would be good